### ISSUE TITLE
Trait in enforced-import-renames should still allow import as _

### ISSUE THREAD

**Issue by taladar** [rust-lang/rust-clippy#16150]
### Summary

Or at the very least this should be a configuration option for the lint. Usually importing a trait this way means the name is never used in the code so it is sensible to not rename e.g. std::io::Write to IoWrite to distinguish it from std::fmt::Write in those cases.

### Lint Name

missing_enforced_import_renames

### Reproducer

I tried this code:

```rust
use std::io::Write as _;
```

with the lint set to deny in Cargo.toml and clippy toml

```toml
enforced-import-renames = [
    { path = "std::fmt::Write", rename = "FmtWrite" },
    { path = "std::io::Write", rename = "IoWrite" },

]
```

I saw this happen:

```
error: this import should be renamed
 --> code.rs:4:1
  |
4 | use std::io::Write as _;
  | ^^^^^^^^^^^^^^^^^^^^^^^ help: try: `use std::io::Write as IoWrite`
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/rust-1.91.0/index.html#missing_enforced_import_renames
```

I expected to see this happen:

No lint error because I assumed the lint was intended for situations where the name is actually imported to be used in the code.

### Version

```text
rustc 1.91.1 (ed61e7d7e 2025-11-07)
binary: rustc
commit-hash: ed61e7d7e242494fb7057f2657300d9e77bb4fcb
commit-date: 2025-11-07
host: x86_64-unknown-linux-gnu
release: 1.91.1
LLVM version: 21.1.2
```

**Comment by sergiogiro** [comment]
I think I can take care of this one.

> Or at the very least this should be a configuration option for the lint.

Do we want for the allowance of "_" to be an option? I think it's unlikely that someone would like to enforce a renaming for a value that it's not used.

@rustbot claim

**Comment by taladar** [comment]
Well, personally I don't think I would ever want forced renaming in those cases, I just meant if it is not changed to avoid a message in those cases unconditionally it should be an option instead of the current behavior where it unconditionally forces the rename even when the name is never used.

**Comment by sergiogiro** [comment]
Understood, thank you.

Also it'd be useful to read what other people think.

**Comment by taladar** [comment]
Do renames affect error messages that mention the trait? If so that would be the only case I can think of where people might still want a rename even though they do not use the name in the code.
